fedimint-api/src/cancellable.rs

```diff
@@ -0,0 +1,9 @@
+use thiserror::Error;
+
+/// An error used as a "cancelled" marker in [`Cancellable`].
+#[derive(Error, Debug)]
+#[error("Operation cancelled")]
+pub struct Cancelled;
+
+/// Operation that can potentially get cancelled returning no result (e.g. program shutdown).
+pub type Cancellable<T> = std::result::Result<T, Cancelled>;
```

Comment:
Any thoughts on this alternative construction where `MaybeCanceled<E>` is essentially an `Option`-like type that implements `From<E>`:

```rust
pub type Cancellable<T, E> = std::result::Result<T, MaybeCancelled<E>>;
```

I think the current design is good at least as good and maybe I'm missing something.

Reply:
https://users.rust-lang.org/t/state-of-art-cancellabe-option-result-or-result-option-handling/83420/10?u=dpc

>  It just occurred to me that whatever OrCancellable<T> wrapper or something is used, it should be wrapping either whole Result or OK part of it, as so it can work with functions that are otherwise non-fallible.

But I'm still unsure about the whole thing :)